lib/uplink/buckets.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"storj.io/storj/pkg/storj"
+	ul "storj.io/storj/uplink"
+)
+
+// BucketOpts holds the cipher, path, key, and enc. scheme for each bucket since they
+// can be different for each
+type BucketOpts struct {
+	PathCipher       storj.Cipher
+	EncPathPrefix    storj.Path
+	Key              storj.Key
+	EncryptionScheme storj.EncryptionScheme
+}
+
+// Bucket is a struct that allows operations on a Bucket after a user providers Permissions
+type Bucket struct {
```

Comment:
should `storj.Bucket` be on this struct? I'm looking at this but not sure how current it is: (https://github.com/storj/storj/pull/847/files)

Reply:
Fixed 👍